aptos-move/e2e-move-tests/src/tests/object_args.data/pack/sources/object_test.move

```diff
@@ -0,0 +1,57 @@
+module 0xCAFE::test {
+    use std::vector;
+    use std::string::String;
+    use std::option::Option;
+    use std::fixed_point32::FixedPoint32;
+    use aptos_std::object::Object;
+    use aptos_framework::object::{create_object_from_account, generate_signer};
+    use aptos_framework::object;
+
+    struct ModuleData has key, store {
+        state: String,
+    }
+
+    struct GenericModuleData<T: copy + store> has key, store {
+        state: T,
+    }
+
+    public entry fun initialize(sender: &signer) {
+        let cref = create_object_from_account(sender);
+        let s = generate_signer(&cref);
+        move_to(&s, ModuleData { state: std::string::utf8(b"") });
+    }
+
+    public entry fun object_arg(msg: String, o: Object<ModuleData>) acquires ModuleData {
+        let addr = aptos_std::object::object_address(&o);
+        // guaranteed to exist
+        borrow_global_mut<ModuleData>(addr).state = msg;
+    }
+
+    public entry fun object_vec(msg: String, objs: vector<Object<ModuleData>>) acquires ModuleData {
+        vector::for_each(objs,|o| { borrow_global_mut<ModuleData>(object::object_address(&o)).state = msg; });
+    }
+
+    public entry fun pass_optional_fixedpoint(o: Object<ModuleData>, x: Option<FixedPoint32>) acquires ModuleData {
```

Comment:
Can you add a test for FixedPoint64 as well?

At this point we should just rename this test to test_struct_constructors as it's testing beyond Objects now

Reply:
I added quite a few cases